app/concepts/movies/repository.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Movies
+  class Repository < ::Base::BaseRepository
+    def initialize(adapter: Movie)
+      super(adapter: adapter)
+    end
```

Comment:
```
def find_movie / def find(id)
  Movie.find(id) #instead of adapter.find(id)
end
```

Reply:
```
def find_todays_movies
Movie.includes(:seanses).where(seans.start_date: todays_scope)
end
```

Reply:
in tests:


```
  repository movie
  
  subject {Movies:: Repository.new.find_todays_movies}  
  let(:expected_repsonse) do
    [
      movie_1, 
      movie_2,
    ]
  end
  it "returns only todays movies " do
    expect(subject).to match(expected_response) 
  end
  
```

Reply:
add before block

```
before do 
movie_1 #-> wiht seans today
movie_2 #-> with seans today and tomorrow
movie_3 #-> with tomorrow seans
end

#instead of movie_1 ~ you can use movie_with_seans_today

Reply:
```
def find_todays_movies
Movie.includes(:seanses).where(seans.start_date: todays_scope)
end

```

it could be also

```
def find_todays_movies
-> from cache
Redis.get("todays_movies")

end
```

Reply:
def promoted_movie
  Movie.find(4)
end

Reply:
> it "returns only todays movies " do
>     expect(subject).to match(expected_response) 
>   end
```
let(:parsed_resutls) do
subject.map { |movie| { id: movie.id, title: movie.title, }  }  
end
let(:expected_response) do 
[{id: movie_2.id, title: movie_2.title}, {id: movie_1.id, title: movie_1.title}]
end
 it "returns only todays movies " do
    expect(parsed_resuts).to match(expected_response) 
  end